examples/platforms/efr32/radio.c

```diff
@@ -760,14 +792,45 @@ static void RAILCb_Generic(RAIL_Handle_t aRailHandle, RAIL_Events_t aEvents)
         status = RAIL_Calibrate(aRailHandle, NULL, RAIL_CAL_ALL_PENDING);
         assert(status == RAIL_STATUS_NO_ERROR);
     }
+
+    if (aEvents & RAIL_EVENT_RSSI_AVERAGE_DONE)
+    {
+        const int16_t energyScanResultQuarterDbm = RAIL_GetAverageRssi(aRailHandle);
+
+        if (energyScanResultQuarterDbm == RAIL_RSSI_INVALID)
+        {
+            sEnergyScanResultDbm = 0;
+            sEnergyScanStatus    = ENERGY_SCAN_COMPLETED_ERROR;
+        }
+        else
+        {
+            sEnergyScanResultDbm = energyScanResultQuarterDbm / QUARTER_DBM_IN_DBM;
+            sEnergyScanStatus    = ENERGY_SCAN_COMPLETED;
+        }
+    }
 }
 
 otError otPlatRadioEnergyScan(otInstance *aInstance, uint8_t aScanChannel, uint16_t aScanDuration)
 {
+    RAIL_Status_t status;
+    otError       error = OT_ERROR_NONE;
+
     (void)aInstance;
-    (void)aScanChannel;
-    (void)aScanDuration;
-    return OT_ERROR_NOT_IMPLEMENTED;
+
+    CORE_DECLARE_IRQ_STATE;
+    CORE_ENTER_CRITICAL();
+
+    otEXPECT_ACTION(((sState == OT_RADIO_STATE_SLEEP) && (RAIL_GetRadioState(sRailHandle) == RAIL_RF_STATE_IDLE) &&
```

Comment:
Is it necessary to require `sState == OT_RADIO_STATE_SLEEP`? OpenThread does not currently assume this constraint and will allow energy scans even while attached to a Thread network.

Reply:
I've removed that check as it was not necessary. `RAIL_GetRadioState()` is the source of truth when deciding whether the radio is in a state to perform the operation.